### ISSUE TITLE
Consul exporter times out on scrape if consul cluster is unhealthy

### ISSUE THREAD

**Issue by tomwilkie** [prometheus/consul_exporter#37]
Which makes it not particularly useful for monitoring the health of your consul cluster...

![screen shot 2016-11-15 at 17 04 01](https://cloud.githubusercontent.com/assets/444037/20315372/87f58762-ab55-11e6-9698-cbc2267eae22.png)

Stack trace from exporter (triggered by `# kill -SIGQUIT 1` in separate terminal)
```
$ kubectl logs -f --namespace=cortex consul-4059865226-9f4fe consul-exporter 
time="2016-11-15T16:46:12Z" level=info msg="Starting consul_exporter (version=, branch=, revision=)" source="consul_exporter.go:285" 
time="2016-11-15T16:46:12Z" level=info msg="Build context (go=go1.7.1, user=, date=)" source="consul_exporter.go:286" 
time="2016-11-15T16:46:12Z" level=info msg="Listening on :9107" source="consul_exporter.go:305" 
SIGQUIT: quit
PC=0x460e69 m=0

goroutine 0 [idle]:
runtime.epollwait(0x7ffe00000004, 0x7ffe32c35f88, 0xffffffff00000080, 0x0, 0xffffffff00000000, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/runtime/sys_linux_amd64.s:525 +0x19
runtime.netpoll(0xa4aa01, 0x0)
	/usr/local/go/src/runtime/netpoll_epoll.go:67 +0x93
runtime.findrunnable(0xc420018000, 0x0)
	/usr/local/go/src/runtime/proc.go:2003 +0x28c
runtime.schedule()
	/usr/local/go/src/runtime/proc.go:2120 +0x14c
runtime.park_m(0xc4201bf1e0)
	/usr/local/go/src/runtime/proc.go:2183 +0x123
runtime.mcall(0x7ffe32c36700)
	/usr/local/go/src/runtime/asm_amd64.s:240 +0x5b

goroutine 1 [IO wait]:
net.runtime_pollWait(0x7f7063f8f0a8, 0x72, 0x0)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420141aa0, 0x72, 0xc420047b58, 0xc420012150)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420141aa0, 0xa25100, 0xc420012150)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).accept(0xc420141a40, 0x0, 0xa23b80, 0xc4202accc0)
	/usr/local/go/src/net/fd_unix.go:419 +0x238
net.(*TCPListener).accept(0xc4200342d8, 0x29e8d60800, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:132 +0x2e
net.(*TCPListener).AcceptTCP(0xc4200342d8, 0xc420047c80, 0xc420047c88, 0xc420047c78)
	/usr/local/go/src/net/tcpsock.go:209 +0x49
net/http.tcpKeepAliveListener.Accept(0xc4200342d8, 0x886c70, 0xc420080400, 0xa28b00, 0xc42015cc90)
	/usr/local/go/src/net/http/server.go:2608 +0x2f
net/http.(*Server).Serve(0xc420080a80, 0xa28300, 0xc4200342d8, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:2273 +0x1ce
net/http.(*Server).ListenAndServe(0xc420080a80, 0xc420080a80, 0x2)
	/usr/local/go/src/net/http/server.go:2219 +0xb4
net/http.ListenAndServe(0x7ffe32c3791e, 0x5, 0x0, 0x0, 0xc420158920, 0x2)
	/usr/local/go/src/net/http/server.go:2351 +0xa0
main.main()
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/consul_exporter.go:306 +0x6a2

goroutine 3080 [select]:
net/http.(*persistConn).roundTrip(0xc4202da000, 0xc42014cfa0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1840 +0x93b
net/http.(*Transport).RoundTrip(0xc4200da0f0, 0xc4201c0000, 0xc4200da0f0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:380 +0x4ee
net/http.send(0xc4201c0000, 0xa234c0, 0xc4200da0f0, 0x0, 0x0, 0x0, 0x8, 0xc42066a130, 0x412a18)
	/usr/local/go/src/net/http/client.go:256 +0x15f
net/http.(*Client).send(0xc42015c0c0, 0xc4201c0000, 0x0, 0x0, 0x0, 0xc42066a130, 0x0, 0x1)
	/usr/local/go/src/net/http/client.go:146 +0x102
net/http.(*Client).doFollowingRedirects(0xc42015c0c0, 0xc4201c0000, 0x886e18, 0x3, 0x582b3e01, 0x337001dc)
	/usr/local/go/src/net/http/client.go:528 +0x5e5
net/http.(*Client).Do(0xc42015c0c0, 0xc4201c0000, 0xa4a880, 0x0, 0xc42023ac00)
	/usr/local/go/src/net/http/client.go:184 +0x1ea
github.com/prometheus/consul_exporter/vendor/github.com/hashicorp/consul/api.(*Client).doRequest(0xc420078960, 0xc420694780, 0x3, 0x8526ee, 0x11, 0xc420694780)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/hashicorp/consul/api/api.go:469 +0x8e
github.com/prometheus/consul_exporter/vendor/github.com/hashicorp/consul/api.(*Catalog).Nodes(0xc420147d98, 0xc420147f00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/hashicorp/consul/api/catalog.go:102 +0xdf
main.(*Exporter).Collect(0xc420156a40, 0xc42023ac00)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/consul_exporter.go:164 +0x2e9
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB.func2(0xc420012700, 0xc42023ac00, 0xa26080, 0xc420156a40)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:415 +0x63
created by github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:416 +0x283

goroutine 2940 [IO wait]:
net.runtime_pollWait(0x7f7063f8ec28, 0x72, 0x7)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc4202e4a70, 0x72, 0xc4200f29d0, 0xc420012150)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc4202e4a70, 0xa25100, 0xc420012150)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc4202e4a10, 0xc4201a6000, 0x1000, 0x1000, 0x0, 0xa25100, 0xc420012150)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc42066a190, 0xc4201a6000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc4202da600, 0xc4201a6000, 0x1000, 0x1000, 0x4d27a0, 0xc4200f2b58, 0x40674d)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc4201ed020)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc4201ed020, 0x1, 0xc4200f2bbd, 0x1, 0x0, 0xc4201ed380, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc4202da600)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 3068 [select]:
net/http.(*persistConn).roundTrip(0xc4202da600, 0xc4202eaf60, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1840 +0x93b
net/http.(*Transport).RoundTrip(0xc4200da0f0, 0xc4200db4a0, 0xc4200da0f0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:380 +0x4ee
net/http.send(0xc4200db4a0, 0xa234c0, 0xc4200da0f0, 0x0, 0x0, 0x0, 0x8, 0xc4206c61a8, 0x412a18)
	/usr/local/go/src/net/http/client.go:256 +0x15f
net/http.(*Client).send(0xc42015c0c0, 0xc4200db4a0, 0x0, 0x0, 0x0, 0xc4206c61a8, 0x0, 0x1)
	/usr/local/go/src/net/http/client.go:146 +0x102
net/http.(*Client).doFollowingRedirects(0xc42015c0c0, 0xc4200db4a0, 0x886e18, 0x3, 0x582b3e01, 0x33a99700)
	/usr/local/go/src/net/http/client.go:528 +0x5e5
net/http.(*Client).Do(0xc42015c0c0, 0xc4200db4a0, 0xa4a880, 0x0, 0xc4201eca20)
	/usr/local/go/src/net/http/client.go:184 +0x1ea
github.com/prometheus/consul_exporter/vendor/github.com/hashicorp/consul/api.(*Client).doRequest(0xc420078960, 0xc4202cebe0, 0x3, 0x8526ee, 0x11, 0xc4202cebe0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/hashicorp/consul/api/api.go:469 +0x8e
github.com/prometheus/consul_exporter/vendor/github.com/hashicorp/consul/api.(*Catalog).Nodes(0xc4205add98, 0xc4205adf00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/hashicorp/consul/api/catalog.go:102 +0xdf
main.(*Exporter).Collect(0xc420156a40, 0xc4201eca20)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/consul_exporter.go:164 +0x2e9
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB.func2(0xc420196880, 0xc4201eca20, 0xa26080, 0xc420156a40)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:415 +0x63
created by github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:416 +0x283

goroutine 3086 [chan send]:
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.computeApproximateRequestSize(0xc4200db2c0, 0xc420556180, 0x44a)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/http.go:187 +0x151
created by github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.InstrumentHandlerFuncWithOpts.func1
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/http.go:146 +0x16a

goroutine 3074 [chan receive]:
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB(0xc4200786c0, 0xa244c0, 0xc4202ac520, 0x0, 0x0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:427 +0x321
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).ServeHTTP(0xc4200786c0, 0xa27a00, 0xc4206b40f0, 0xc4200da1e0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:375 +0x13b
net/http.(Handler).ServeHTTP-fm(0xa27a00, 0xc4206b40f0, 0xc4200da1e0)
	/usr/local/go/src/net/http/h2_bundle.go:4084 +0x4d
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.InstrumentHandlerFuncWithOpts.func1(0xa27f40, 0xc4201b2000, 0xc4200da1e0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/http.go:158 +0x2ab
net/http.HandlerFunc.ServeHTTP(0xc42000f450, 0xa27f40, 0xc4201b2000, 0xc4200da1e0)
	/usr/local/go/src/net/http/server.go:1726 +0x44
net/http.(*ServeMux).ServeHTTP(0xa4a640, 0xa27f40, 0xc4201b2000, 0xc4200da1e0)
	/usr/local/go/src/net/http/server.go:2022 +0x7f
net/http.serverHandler.ServeHTTP(0xc420080a80, 0xa27f40, 0xc4201b2000, 0xc4200da1e0)
	/usr/local/go/src/net/http/server.go:2202 +0x7d
net/http.(*conn).serve(0xc420972100, 0xa28a40, 0xc42067e280)
	/usr/local/go/src/net/http/server.go:1579 +0x4b7
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2293 +0x44d

goroutine 2941 [select]:
net/http.(*persistConn).writeLoop(0xc4202da600)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 3024 [IO wait]:
net.runtime_pollWait(0x7f7063f8e868, 0x72, 0xc)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc4202e46f0, 0x72, 0xc4206b29d0, 0xc420012150)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc4202e46f0, 0xa25100, 0xc420012150)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc4202e4690, 0xc4202be000, 0x1000, 0x1000, 0x0, 0xa25100, 0xc420012150)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc4206c6080, 0xc4202be000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc4202da000, 0xc4202be000, 0x1000, 0x1000, 0x4d27a0, 0xc4206b2b58, 0x40674d)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc4202bc960)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc4202bc960, 0x1, 0xc4206b2bbd, 0x1, 0x0, 0xc42023aea0, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc4202da000)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 3090 [select]:
net/http.(*persistConn).roundTrip(0xc4202da500, 0xc4202eae40, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1840 +0x93b
net/http.(*Transport).RoundTrip(0xc4200da0f0, 0xc4200db3b0, 0xc4200da0f0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:380 +0x4ee
net/http.send(0xc4200db3b0, 0xa234c0, 0xc4200da0f0, 0x0, 0x0, 0x0, 0x8, 0xc4206c6180, 0x412a18)
	/usr/local/go/src/net/http/client.go:256 +0x15f
net/http.(*Client).send(0xc42015c0c0, 0xc4200db3b0, 0x0, 0x0, 0x0, 0xc4206c6180, 0x0, 0x1)
	/usr/local/go/src/net/http/client.go:146 +0x102
net/http.(*Client).doFollowingRedirects(0xc42015c0c0, 0xc4200db3b0, 0x886e18, 0x3, 0x582b3e01, 0x33a7724b)
	/usr/local/go/src/net/http/client.go:528 +0x5e5
net/http.(*Client).Do(0xc42015c0c0, 0xc4200db3b0, 0xa4a880, 0x0, 0xc4205561e0)
	/usr/local/go/src/net/http/client.go:184 +0x1ea
github.com/prometheus/consul_exporter/vendor/github.com/hashicorp/consul/api.(*Client).doRequest(0xc420078960, 0xc4202ceb90, 0x3, 0x8526ee, 0x11, 0xc4202ceb90)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/hashicorp/consul/api/api.go:469 +0x8e
github.com/prometheus/consul_exporter/vendor/github.com/hashicorp/consul/api.(*Catalog).Nodes(0xc420149d98, 0xc420149f00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/hashicorp/consul/api/catalog.go:102 +0xdf
main.(*Exporter).Collect(0xc420156a40, 0xc4205561e0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/consul_exporter.go:164 +0x2e9
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB.func2(0xc4206be730, 0xc4205561e0, 0xa26080, 0xc420156a40)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:415 +0x63
created by github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:416 +0x283

goroutine 3064 [chan send]:
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.computeApproximateRequestSize(0xc4201c00f0, 0xc4201ec9c0, 0x44a)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/http.go:187 +0x151
created by github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.InstrumentHandlerFuncWithOpts.func1
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/http.go:146 +0x16a

goroutine 3073 [select]:
net/http.(*persistConn).writeLoop(0xc4202da000)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 2971 [chan receive]:
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB(0xc4200786c0, 0xa244c0, 0xc4202eaa60, 0x0, 0x0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:427 +0x321
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).ServeHTTP(0xc4200786c0, 0xa27a00, 0xc420034178, 0xc4201c00f0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:375 +0x13b
net/http.(Handler).ServeHTTP-fm(0xa27a00, 0xc420034178, 0xc4201c00f0)
	/usr/local/go/src/net/http/h2_bundle.go:4084 +0x4d
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.InstrumentHandlerFuncWithOpts.func1(0xa27f40, 0xc4201b2f70, 0xc4201c00f0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/http.go:158 +0x2ab
net/http.HandlerFunc.ServeHTTP(0xc42000f450, 0xa27f40, 0xc4201b2f70, 0xc4201c00f0)
	/usr/local/go/src/net/http/server.go:1726 +0x44
net/http.(*ServeMux).ServeHTTP(0xa4a640, 0xa27f40, 0xc4201b2f70, 0xc4201c00f0)
	/usr/local/go/src/net/http/server.go:2022 +0x7f
net/http.serverHandler.ServeHTTP(0xc420080a80, 0xa27f40, 0xc4201b2f70, 0xc4201c00f0)
	/usr/local/go/src/net/http/server.go:2202 +0x7d
net/http.(*conn).serve(0xc420080400, 0xa28a40, 0xc4206aacc0)
	/usr/local/go/src/net/http/server.go:1579 +0x4b7
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2293 +0x44d

goroutine 3087 [semacquire]:
sync.runtime_Semacquire(0xc4206be73c)
	/usr/local/go/src/runtime/sema.go:47 +0x30
sync.(*WaitGroup).Wait(0xc4206be730)
	/usr/local/go/src/sync/waitgroup.go:131 +0x97
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB.func1(0xc4206be730, 0xc4205561e0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:409 +0x2d
created by github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:411 +0x1db

goroutine 3075 [chan send]:
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.computeApproximateRequestSize(0xc4200da1e0, 0xc42023aba0, 0x44a)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/http.go:187 +0x151
created by github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.InstrumentHandlerFuncWithOpts.func1
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/http.go:146 +0x16a

goroutine 3056 [IO wait]:
net.runtime_pollWait(0x7f7063f8e9e8, 0x72, 0x8)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420141f70, 0x72, 0xc4200f59d0, 0xc420012150)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420141f70, 0xa25100, 0xc420012150)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc420141f10, 0xc42011e000, 0x1000, 0x1000, 0x0, 0xa25100, 0xc420012150)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc4206c6160, 0xc42011e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc4202da500, 0xc42011e000, 0x1000, 0x1000, 0x4d27a0, 0xc4200f5b58, 0x40674d)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc4202d0cc0)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc4202d0cc0, 0x1, 0xc4200f5bbd, 0x1, 0x0, 0xc4201ed200, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc4202da500)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 3121 [select]:
net/http.(*persistConn).writeLoop(0xc4202da500)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 3065 [semacquire]:
sync.runtime_Semacquire(0xc42019688c)
	/usr/local/go/src/runtime/sema.go:47 +0x30
sync.(*WaitGroup).Wait(0xc420196880)
	/usr/local/go/src/sync/waitgroup.go:131 +0x97
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB.func1(0xc420196880, 0xc4201eca20)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:409 +0x2d
created by github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:411 +0x1db

goroutine 3085 [chan receive]:
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB(0xc4200786c0, 0xa244c0, 0xc4202acc00, 0x0, 0x0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:427 +0x321
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).ServeHTTP(0xc4200786c0, 0xa27a00, 0xc4206b4140, 0xc4200db2c0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:375 +0x13b
net/http.(Handler).ServeHTTP-fm(0xa27a00, 0xc4206b4140, 0xc4200db2c0)
	/usr/local/go/src/net/http/h2_bundle.go:4084 +0x4d
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.InstrumentHandlerFuncWithOpts.func1(0xa27f40, 0xc4201b2410, 0xc4200db2c0)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/http.go:158 +0x2ab
net/http.HandlerFunc.ServeHTTP(0xc42000f450, 0xa27f40, 0xc4201b2410, 0xc4200db2c0)
	/usr/local/go/src/net/http/server.go:1726 +0x44
net/http.(*ServeMux).ServeHTTP(0xa4a640, 0xa27f40, 0xc4201b2410, 0xc4200db2c0)
	/usr/local/go/src/net/http/server.go:2022 +0x7f
net/http.serverHandler.ServeHTTP(0xc420080a80, 0xa27f40, 0xc4201b2410, 0xc4200db2c0)
	/usr/local/go/src/net/http/server.go:2202 +0x7d
net/http.(*conn).serve(0xc420080100, 0xa28a40, 0xc4206aa780)
	/usr/local/go/src/net/http/server.go:1579 +0x4b7
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2293 +0x44d

goroutine 3076 [semacquire]:
sync.runtime_Semacquire(0xc42001270c)
	/usr/local/go/src/runtime/sema.go:47 +0x30
sync.(*WaitGroup).Wait(0xc420012700)
	/usr/local/go/src/sync/waitgroup.go:131 +0x97
github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB.func1(0xc420012700, 0xc42023ac00)
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:409 +0x2d
created by github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*registry).writePB
	/Users/twilkie/Documents/src/github.com/prometheus/consul_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:411 +0x1db

rax    0xfffffffffffffffc
rbx    0xa4aa68
rcx    0x460e69
rdx    0x80
rdi    0x4
rsi    0x7ffe32c35f88
rbp    0x7ffe32c36588
rsp    0x7ffe32c35f48
r8     0x0
r9     0x0
r10    0xffffffff
r11    0x246
r12    0x8
r13    0x188
r14    0x32
r15    0x3
rip    0x460e69
rflags 0x246
cs     0x33
fs     0x0
gs     0x0
```

Looks to me like its blockoing on this line:

```go
	// How many nodes are registered?
	nodes, _, err := e.client.Catalog().Nodes(&consul_api.QueryOptions{})
```

**Comment by tomwilkie** [comment]
Should note, this is with consul 0.7.1

**Comment by tomwilkie** [comment]
Fixed by #36
